src/main/java/racingcar/domain/input/InputRoundCount.java

```diff
@@ -0,0 +1,15 @@
+package racingcar.domain.input;
+
+public class InputRoundCount extends PositiveNumber {
+
+    private int roundCount;
+
+    public InputRoundCount(String value) {
+        super(value);
+        this.roundCount = super.positiveNumber;
+    }
```

Comment:
```suggestion
public class InputRoundCount {

    private PositiveNumber roundCount;

    public InputRoundCount(PositiveNumber roundCount) {
        this.roundCount = roundCount;
    }
```

저는 이럴 경우에는 상속을 하기 보다는 "원시값인 int 를 PositiveNumber 로 포장하는 방식"을 좋아합니다.

> getCarCount 메소드에서 리턴을 return super.positiveNumber; 이 더 좋은 코드일까요?

저는 웅래님처럼 상속을 하게 되는 방식을 매우우우우 싫어합니다.
매우 제 취향이 아닙니다.

아마도 IS-A 관계에 따라서,
RoundCount 는 PositiveNumber 이지만,
PositiveNumber 는 RoundCount 가 아니다!
라는 관점에서 이러한 상속관계를 설계하신것 같습니다.

즉 웅래님의 방식은 충분히 합리적인 이유가 있고, 틀리지 않았습니다.
그리고 roundCount 를 따로 명시하는 편이 가독성이 좋다는 의견또한 저는 맞다고 생각합니다.

하지만 일단 제 취향에서 왜 위 방식을 좋아하는지 않는지 한번 말씀드려보겠습니다.
일단 위에서 positiveNumber 와 roundCount 는 항상 같은 값이 되어야 하는게 개발 의도입니다.
하지만 roundCount 와 positiveNumber 가 별도의 멤버로써 분리되어 있습니다.
즉 this.roundCount 와 super.positiveNumber 는 컴퓨터 메모리상으로 전혀 다른 값입니다.
다른 말로 의미해서, 이 둘의 값이 달라질 가능성이 있다는 것을 뜻합니다.

웅래님이 작성하실 때는 모르지만, 웅래님의 코드를 다른 사람이 협업해서 사용하게 될때는 이야기가 달라집니다.
어떤 기상천외한 실수가 발생할지 모르고, 그럴 가능성을 처음부터 막아두는 것을 좋아합니다.

그렇게 해서 만약 roundCount 라는 멤버를 없애고,
`return super.positiveNumber;` 를 하게 되면, InputRoundCount 라는 클래스의 역할이 거의 사라지는 것을 볼 수 있습니다.

즉 누가 보기에도 InputRoundCount 는 하는일이 없는 
역할과 책임이 없는 아이 입니다.

Reply:
협업해서 사용될때 기상천외한 실수... 넵넵 너무 공감합니다ㅜㅜ
그럴 가능성이 있으면 처음부터 막는 것... 좋은 피드백 감사합니다
메모리상의 전혀 다른 값이여서 final로 막으려는 의도가 있었는데 roundCount의 경우 final이 없어 
positiveNumber와 roundCount값이 달라질 수 있고, 협업을 하다보면 생각도 못한 이슈가 발생될 수 가 있겠네요...
좋은 지적감사합니다!